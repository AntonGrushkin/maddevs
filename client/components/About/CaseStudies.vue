<template>
  <section
    id="case-studies"
    class="case-studies"
    :class="{ 'case-studies--projects': currentPageName === 'projects' }"
  >
    <div class="container">
      <div>
        <h2
          class="case-studies__main-title main-title"
          :class="{
            'case-studies--projects__main-title': currentPageName === 'projects'
          }"
        >
          Case
          <span>Studies</span>
        </h2>
        <img
          class="projects-flame-background"
          loading="lazy"
          src="@/assets/img/Projects/png/projects-flame-bg.png"
          alt="Projects Flame Image"
          v-if="currentPageName === 'projects'"
        />
      </div>
      <div class="projects">
        <SingleProject
          v-for="(project, i) in projects"
          :key="i"
          :project="project"
        />
      </div>
    </div>
    <div class="projects-mobile-slider">
      <swiper class="swiper" :options="swiperOption">
        <swiper-slide v-for="(project, i) in projects" :key="i">
          <SingleProject :project="project" />
        </swiper-slide>
      </swiper>
      <div class="swiper-pagination swiper-pagination--white" slot="pagination"/>
    </div>
  </section>
</template>

<script>
import { Swiper, SwiperSlide } from 'vue-awesome-swiper';
import 'swiper/css/swiper.css';
import SingleProject from '@/components/About/SingleProject';

export default {
  name: 'CaseStudies',
  components: {
    Swiper,
    SwiperSlide,
    SingleProject
  },
  beforeMount() {
    // Set new params for slider
    if (window.screen.width <= 480) {
      this.swiperOption.slidesPerView = 1.05;
    }
  },
  mounted() {
    if ($nuxt.$route.name) {
      this.currentPageName = $nuxt.$route.name;
    }
  },
  data() {
    return {
      swiperOption: {
        pagination: {
          el: '.swiper-pagination--white'
        },
        slidesPerView: 1.05
      },
      currentPageName: '',
      projects: [
        {
          projectName: 'nambafood',
          logoImg: 'nambafoodColored',
          link: 'https://blog.maddevs.io/namba-food-top-1-delivery-service-in-central-asia-3df8ca4d436d?source=friends_link&sk=1dd67cc82dad8e4fae756b3a4d47f121',
          contributionWidgetColors: ['#ec1c24', '#96969c'],
          projectBackground: 'nambafoodBackground',
          projectColor: '#f8f7f6',
          projectTitle: 'Contactless Delivery Service',
          projectDescription:
            'Mad Devs created the Namba Food delivery service from scratch. The apps for couriers, end users placing orders and business owners work as a seamless system, ensuring a smooth delivery process for food and other goods.'
        },
        {
          projectName: 'teacherly',
          logoImg: 'teacherlyColored',
          link: 'https://blog.maddevs.io/teacherly-provides-collaboration-platform-to-70-000-teachers-88631cc5195f?source=friends_link&sk=0f492513f43ce5b4330c3cfa484378c5',
          contributionWidgetColors: ['#ec1c24', '#fff'],
          projectBackground: 'teacherlyBackground',
          projectColor: '#56448e',
          projectTitle: 'EdTech collaboration platform',
          projectDescription:
            'Mad Devs improved the collaboration experience for teachers and students by the feedback-driven development of the Teacherly educational platform.'
        },
        {
          projectName: 'guardrails',
          logoImg: 'guardrailsColored',
          link: 'https://blog.maddevs.io/guardrails-helps-800-development-teams-to-establish-devsecops-with-automated-security-reviews-806c3c1d516b?source=friends_link&sk=15f4cdb98f9410b27b9d76dfd94ad1d0',
          contributionWidgetColors: ['#96969c', '#96969c'],
          projectBackground: 'guardrailsBackground',
          projectColor: '#0e1b27',
          projectTitle: 'Cloud cybersecurity service',
          projectDescription:
            'Mad Devs was involved with Guardrails\' security check service as a development contractor with exceptional knowledge of GitHub and GitLab processes.'
        },
        {
          projectName: 'godee',
          logoImg: 'godeeColored',
          link: '/godee-case-study',
          contributionWidgetColors: ['#000', '#000'],
          projectBackground: 'godeeBackground',
          projectColor: '#ff6A01',
          projectTitle: 'Convenient shuttle bus service',
          projectDescription: 'Mad Devs helped GoDee with developing feature-rich software to re-invent public mobility by building new smart ways of a daily commute.'
        }
      ]
    };
  }
};
</script>

<style lang="scss" scoped>
@import '../../assets/styles/vars';

.swiper-container {
  background: $bgcolor--project-white;
}

.case-studies {

  .container {
    position: relative;
  }

  &__main-title {
    padding-bottom: 2vw;
    display: inline-block;
    color: $text-color--red;

    span {
      color: $text-color--black;
      -webkit-text-stroke: $text-stroke--red;
    }
  }

  .projects {
    display: flex;
    flex-wrap: wrap;

    &-flame-background {
      position: absolute;
      bottom: 85vw;
      right: 7vw;
      width: 26vw;
    }

    .single-project {
      &__background_first-project {
        background-size: cover;
      }

      &:nth-child(3) {

        &::before {
          content: '';
          position: absolute;
          background: url('../../assets/img/Studies/svg/guardrailsSecondaryBg.svg') no-repeat right 50%;
          width: 100%;
          height: 25vw;
          background-size: contain;
          z-index: -44;
          top: 0;
          right: 0;
          mix-blend-mode: overlay;
          opacity: 0.68;
        }
      }

      &:last-child {

        &::after {
          content: '';
          position: absolute;
          background-image: url('../../assets/img/Studies/jpg/godeeSecondaryBg.jpg');
          background-repeat: no-repeat;
          width: 100%;
          height: 100%;
          background-size: contain;
          z-index: -44;
          top: 0;
          right: 0;
        }
      }
    }
  }

  .swiper-container {
    background: $bgcolor--black;
  }
}

.projects-mobile-slider {
  display: none;
}

@media only screen and (max-width: 1600px) {
  .case-studies {
    &__main-title {
      font-size: 8vw;
    }
  }
}

@media only screen and (max-width: 745px) {
  .case-studies {
    .projects {
      display: none;
    }

    .projects-flame-background {
      bottom: -5vw;
      right: 7vw;
      width: 22vw;
    }

    .projects-mobile-slider {
      display: block;

      .swiper-slide {
        &:nth-child(3) {
          .single-project {
            &::before {
              content: '';
              position: absolute;
              background: url('../../assets/img/Studies/svg/guardrailsSecondaryBg.svg') no-repeat right 50%;
              width: 100%;
              height: 35vw;
              background-size: contain;
              z-index: -44;
              top: 0;
              right: 0;
              mix-blend-mode: overlay;
              opacity: 0.68;
            }
          }
        }

        &:last-child {
          .single-project {
            &::after {
              content: '';
              position: absolute;
              background-image: url('../../assets/img/Studies/jpg/godeeSecondaryBg.jpg');
              background-repeat: no-repeat;
              width: 100%;
              height: 100%;
              background-size: contain;
              z-index: -44;
              top: 0;
              right: 0;
            }
          }
        }
      }

      .swiper {
        display: block;
      }
    }

    .swiper-slide {
      height: auto;

      .single-project {
        width: 100%;
        min-height: 100%;
      }
    }
  }
}

@media only screen and (max-width: 640px) {
  .case-studies {
    &__main-title {
      font-size: 11vw;
      padding-bottom: 8vw;
    }
  }
}
</style>
