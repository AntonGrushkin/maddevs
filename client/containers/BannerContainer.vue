<template>
    <div class="container banner-container">
      <AboutBanner v-if="currentPage === 'index'"/>
      <ServicesBanner v-if="currentPage === 'services'"/>
      <ProjectsBanner v-if="currentPage === 'projects'"/>
      <CareersBanner v-if="currentPage === 'careers'"/>
    </div>
</template>

<script>
import AboutBanner from '@/components/Banner/AboutBanner';
import ServicesBanner from '@/components/Banner/ServicesBanner';
import ProjectsBanner from '@/components/Banner/ProjectsBanner';
import CareersBanner from '@/components/Banner/CareersBanner';

export default {
  name: 'BannerContainer',
  components: {
    AboutBanner,
    ServicesBanner,
    ProjectsBanner,
    CareersBanner
  },
  mounted() {
    if ($nuxt.$route.name) {
      this.currentPage = $nuxt.$route.name;
    }
  },
  data() {
    return {
      currentPage: null
    };
  }
};
</script>

<style lang="scss">
  @import '../assets/styles/vars';

  .banner-container {
    width: 100%;
    height: 100%;
    position: relative;
    background-color: $bgcolor--black;

    .banner-content {
      width: 100%;
      height: 100%;
      padding-top: 190px;
    }

    .banner-main_title,
    .banner-main_title--white,
    .banner-main_title--white-first {
      font-family: 'Hoves-Bold';
      font-style: normal;
      font-weight: bold;
      font-size: 10.4vw;
      color: $text-color--black;
      -webkit-text-stroke: $text-stroke--white;
      line-height: 100%;
      letter-spacing: -9px;

      .main-title-white_part {
        color: $text-color--white;
        -webkit-text-stroke: 0;
      }

      &--white,
      &--white-first {
        color: $text-color--white;
        -webkit-text-stroke: 0;
      }
    }

    .text-content_wrapper {
      display: flex;
      padding-bottom: 12vw;
    }
  }

  @media only screen and (max-width: 1024px) {
    .banner-container {
      .banner-content {
        padding-top: 0;
      }
    }
  }

  @media only screen and (max-width: 610px) {
    .banner-container {
      .banner-main_title--white-first {
        display: inline;
      }
    }
  }

  @media only screen and (max-width: 755px) {
    .banner-container {
      .banner-main_title,
      .banner-main_title--white,
      .banner-main_title--white-first {
        font-size: 15vw;
        letter-spacing: -2px;
        line-height: 17vw;
        padding-bottom: 23vw;
      }
    }
  }
</style>
