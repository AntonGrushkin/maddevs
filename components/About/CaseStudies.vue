<template>
  <section class="case-studies">
    <div class="container">
      <h2 class="case-studies__main-title main-title">
        Case
        <span>Studies</span>
      </h2>
      <div class="projects">
        <SingleProject v-for="(project, i) in projects" :key="i" :project="project"/>
      </div>
    </div>
    <div class="projects-mobile-slider">
      <swiper class="swiper">
        <swiper-slide v-for="(project, i) in projects" :key="i">
          <SingleProject :project="project" />
        </swiper-slide>
      </swiper>
    </div>
  </section>
</template>

<script>
import { Swiper, SwiperSlide } from 'vue-awesome-swiper';
import 'swiper/css/swiper.css';
import SingleProject from '@/components/About/SingleProject';

export default {
  name: 'CaseStudies',
  components: {
    Swiper,
    SwiperSlide,
    SingleProject
  },
  data() {
    return {
      projects: [
        {
          projectName: 'nambafood',
          logoImg: 'nambafoodColored',
          contributionVidgetColors: ['#ec1c24', '#96969c'],
          projectBackground: 'nambafoodBackground',
          projectColor: '#f8f7f6',
          projectTitle: 'Contactless delivery service',
          projectDescription: 'Mad Devs created the Namba Food delivery service from scratch. The apps for couriers, end users placing orders and business owners work as a seamless system, ensuring a smooth delivery process for food and other goods.'
        },
        {
          projectName: 'teacherly',
          logoImg: 'teacherlyColored',
          contributionVidgetColors: ['#ec1c24', '#fff'],
          projectBackground: 'teacherlyBackground',
          projectColor: '#56448e',
          projectTitle: 'EdTech collaboration platform',
          projectDescription: 'Mad Devs improved the collaboration experience for teachers and students by the feedback-driven development of the Teacherly educational platform.'
        },
        {
          projectName: 'guardrails',
          logoImg: 'guardrailsColored',
          contributionVidgetColors: ['#96969c', '#96969c'],
          projectBackground: 'guardrailsBackground',
          projectColor: '#0e1b27',
          projectTitle: 'Cloud cybersecurity service',
          projectDescription: 'Mad Devs was involved with Guardrails\' security check service as a development contractor with exceptional knowledge of GitHub and GitLab processes.'
        },
        {
          projectName: 'godee',
          logoImg: 'godeeColored',
          contributionVidgetColors: ['#000', '#000'],
          projectBackground: 'godeeBackground',
          projectColor: '#ff6A01',
          projectTitle: 'Mass transportation company',
          projectDescription: 'Mad Devs helped to automate bus transportation in Vietnam by creating feature-rich GoDee applications for both commuters and bus drivers.'
        }
      ]
    };
  }
};
</script>

<style lang="scss" scoped>
@import '../../assets/styles/vars';

.swiper-container {
  background: $bgcolor--project-white;
}

.case-studies {
  &__main-title {
    padding-bottom: 41px;
    display: inline-block;
    color: $text-color--red;

    span {
      color: $text-color--black;
      text-shadow: $text-shadow--red;
    }
  }

  .projects {
    display: flex;
    flex-wrap: wrap;

    .single-project {
      &:nth-child(3) {
        &::before {
          content: '';
          position: absolute;
          background: url('../../assets/img/Studies/svg/guardrailsSecondaryBg.svg') no-repeat right 50%;
          width: 100%;
          height: 430px;
          background-size: contain;
          z-index: -44;
          top: 0;
          right: 0;
          mix-blend-mode: overlay;
          opacity: 0.68;
        }
      }

      &:last-child {
        &::after {
          content: '';
          position: absolute;
          background-image: url('../../assets/img/Studies/svg/godeeSecondaryBg.svg');
          background-repeat: no-repeat;
          width: 100%;
          height: 100%;
          background-size: contain;
          z-index: -44;
          top: 0;
          right: 0;
        }
      }
    }
  }

  .swiper-container {
    background: $bgcolor--black;
  }

  .swiper-slide {
    &:nth-child(3) {
      .single-project {
        &::before {
          content: '';
          position: absolute;
          background: url('../../assets/img/Studies/svg/guardrailsSecondaryBg.svg') no-repeat right 50%;
          width: 100%;
          height: 310px;
          background-size: contain;
          z-index: -44;
          top: 0;
          right: 0;
          mix-blend-mode: overlay;
          opacity: 0.68;
        }
      }
    }

    &:last-child {
      .single-project {
        &::after {
          content: '';
          position: absolute;
          background-image: url('../../assets/img/Studies/svg/godeeSecondaryBg.svg');
          background-repeat: no-repeat;
          width: 100%;
          height: 100%;
          background-size: contain;
          z-index: -44;
          top: 0;
          right: 0;
        }
      }
    }
  }

  @media only screen and (max-width: 520px) {
    .case-studies {
      &__main-title {
        padding-bottom: 31px;
      }
    }
  }
}

.projects-mobile-slider {
  display: none;
}

@media only screen and (max-width: 768px) {
  .case-studies {

    &__main-title {
      padding-bottom: 31px;
    }

    .projects {
      display: none;
    }

    .projects-mobile-slider {
      display: block;
    }

    .swiper-slide {
      left: -16px;
    }
  }
}
</style>
