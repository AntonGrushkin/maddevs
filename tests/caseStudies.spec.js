import {
  mount
} from '@vue/test-utils';
import CaseStudies from '@/components/About/CaseStudies';
import BannerContainer from '@/containers/BannerContainer';

describe('CaseStudies component', () => {
  let wrapper;
  const projects = [
    {
      projectName: 'nambafood',
      logoImg: 'nambafoodColored',
      link: 'https://blog.maddevs.io/namba-food-startup-e6564e66806d',
      contributionWidgetColors: ['#ec1c24', '#96969c'],
      projectBackground: 'nambafoodBackground',
      projectColor: '#f8f7f6',
      projectTitle: 'Contactless Delivery Service',
      projectDescription:
        'Mad Devs created the Namba Food delivery service from scratch. The apps for couriers, end users placing orders and business owners work as a seamless system, ensuring a smooth delivery process for food and other goods.'
    },
    {
      projectName: 'teacherly',
      logoImg: 'teacherlyColored',
      link: 'https://blog.maddevs.io/teacherly-provides-collaboration-platform-to-70-000-teachers-88631cc5195f',
      contributionWidgetColors: ['#ec1c24', '#fff'],
      projectBackground: 'teacherlyBackground',
      projectColor: '#56448e',
      projectTitle: 'EdTech collaboration platform',
      projectDescription:
        'Mad Devs improved the collaboration experience for teachers and students by the feedback-driven development of the Teacherly educational platform.'
    },
    {
      projectName: 'guardrails',
      logoImg: 'guardrailsColored',
      link: 'https://blog.maddevs.io/guardrails-helps-800-development-teams-to-establish-devsecops-with-automated-security-reviews-806c3c1d516b',
      contributionWidgetColors: ['#96969c', '#96969c'],
      projectBackground: 'guardrailsBackground',
      projectColor: '#0e1b27',
      projectTitle: 'Cloud cybersecurity service',
      projectDescription:
        'Mad Devs was involved with Guardrails\' security check service as a development contractor with exceptional knowledge of GitHub and GitLab processes.'
    },
    {
      projectName: 'godee',
      logoImg: 'godeeColored',
      link: 'https://blog.maddevs.io/building-new-public-transport-habits-in-vietnam-with-godee-f7c4da25efc5',
      contributionWidgetColors: ['#000', '#000'],
      projectBackground: 'godeeBackground',
      projectColor: '#ff6A01',
      projectTitle: 'Mass transportation company',
      projectDescription:
        'Mad Devs helped to automate bus transportation in Vietnam by creating feature-rich GoDee applications for both commuters and bus drivers.'
    }
  ];

  beforeEach(() => {
    const $route = {
      path: '/'
    };

    global.$nuxt = {
      $route: {
        name: '/'
      }
    };

    wrapper = mount(CaseStudies, {
      mocks: {
        $route
      }
    });
  });

  test('is Vue\'s instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('correctly sets the current page\'s default name', () => {
    expect(wrapper.vm.$data.currentPageName).toBe('/');
  });

  test('correctly sets the current page\'s name', () => {
    const $route = {
      path: '/projects'
    };

    wrapper = mount(CaseStudies, {
      mocks: {
        $route
      }
    });

    wrapper.vm.$data.currentPageName = 'projects';
    expect(wrapper.vm.$route.path).toBe('/projects');
    expect(wrapper.vm.$data.currentPageName).toBe('projects');
  });

  test('should have Banner parent component ', () => {
    const $route = {
      path: '/projects'
    };

    wrapper = mount(CaseStudies, {
      mocks: {
        $route
      },
      parentComponent: BannerContainer
    });

    expect(wrapper.vm.$parent.$options.name).toBe('Banner');
  });

  test('should check existence of data', () => {
    expect(wrapper.vm.$data.projects).toStrictEqual(projects);
  });

  test('should render mobile slider', () => {
    let mobileSlider = wrapper.findAll('.projects-mobile-slider');
    expect(mobileSlider).toHaveLength(1);
  });

  test('should render 4 single-projects in the project section', () => {
    let singleProjects = wrapper.findAll('.projects .single-project');
    expect(singleProjects).toHaveLength(4);
  });

  test('should have 16 projects including projects in the slider', () => {
    let sliderSingleProjects = wrapper.findAll('.single-project');
    expect(sliderSingleProjects).toHaveLength(8);
  });

  test('check that the specific images exists', () => {
    const img = wrapper.findAll('.single-project__logo');
    expect(img.length).toBe(8);
  });
});
