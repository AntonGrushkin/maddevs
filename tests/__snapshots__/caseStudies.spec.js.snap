// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaseStudies component renders correctly 1`] = `
<section
  class="case-studies"
>
  <div
    class="container"
  >
    <h2
      class="case-studies__main-title main-title"
    >
      
      Case
      
      <span>
        Studies
      </span>
    </h2>
     
    <div
      class="projects"
    >
      <div
        class="single-project"
        style="background: rgb(248, 247, 246);"
      >
        <div
          class="single-project__container"
        >
          <div
            class="single-project__content-wrap"
          >
            <img
              alt="Namba Food"
              class="single-project__logo"
              src="[object Object]"
            />
             
            <h4
              class="single-project__sub-title sub-title"
            >
              Contactless delivery service
            </h4>
             
            <p
              class="single-project__paragraph paragraph"
            >
              Mad Devs created the Namba Food delivery service from scratch. The
      apps for couriers, end users placing orders and business owners
      work as a seamless system, ensuring a smooth delivery process for
      food and other goods.
            </p>
             
            <div
              class="contribution-vidget"
            >
              <img
                alt="Mad Devs"
                src="[object Object]"
              />
               
              <span
                class="contribution-vidget__content"
              >
                contribution: backend, infrastructure
              </span>
            </div>
          </div>
           
          <div
            class="single-project__background"
          />
        </div>
      </div>
       
      <div
        class="single-project"
        style="background: rgb(86, 68, 142);"
      >
        <div
          class="single-project__container"
        >
          <div
            class="single-project__content-wrap"
          >
            <img
              alt="Namba Food"
              class="single-project__logo"
              src="[object Object]"
            />
             
            <h4
              class="single-project__sub-title sub-title"
            >
              EdTech collaboration platform
            </h4>
             
            <p
              class="single-project__paragraph paragraph"
            >
              Mad Devs improved the collaboration experience for teachers and students by the feedback-driven development of the Teacherly educational platform.
            </p>
             
            <div
              class="contribution-vidget"
            >
              <img
                alt="Mad Devs"
                src="[object Object]"
              />
               
              <span
                class="contribution-vidget__content"
              >
                contribution: backend, infrastructure
              </span>
            </div>
          </div>
           
          <div
            class="single-project__background"
          />
        </div>
      </div>
       
      <div
        class="single-project"
        style="background: rgb(14, 27, 39);"
      >
        <div
          class="single-project__container"
        >
          <div
            class="single-project__content-wrap"
          >
            <img
              alt="Namba Food"
              class="single-project__logo"
              src="[object Object]"
            />
             
            <h4
              class="single-project__sub-title sub-title"
            >
              Cloud cybersecurity service
            </h4>
             
            <p
              class="single-project__paragraph paragraph"
            >
              Mad Devs was involved with Guardrails' security check service as a development contractor with exceptional knowledge of GitHub and GitLab processes.
            </p>
             
            <div
              class="contribution-vidget"
            >
              <img
                alt="Mad Devs"
                src="[object Object]"
              />
               
              <span
                class="contribution-vidget__content"
              >
                contribution: backend, infrastructure
              </span>
            </div>
          </div>
           
          <div
            class="single-project__background"
          />
        </div>
      </div>
       
      <div
        class="single-project"
        style="background: rgb(255, 106, 1);"
      >
        <div
          class="single-project__container"
        >
          <div
            class="single-project__content-wrap"
          >
            <img
              alt="Namba Food"
              class="single-project__logo"
              src="[object Object]"
            />
             
            <h4
              class="single-project__sub-title sub-title"
            >
              Mass transportation company
            </h4>
             
            <p
              class="single-project__paragraph paragraph"
            >
              Mad Devs helped to automate bus transportation in Vietnam by creating feature-rich GoDee applications for both commuters and bus drivers.
            </p>
             
            <div
              class="contribution-vidget"
            >
              <img
                alt="Mad Devs"
                src="[object Object]"
              />
               
              <span
                class="contribution-vidget__content"
              >
                contribution: backend, infrastructure
              </span>
            </div>
          </div>
           
          <div
            class="single-project__background"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="projects-mobile-slider"
  >
    <div
      class="swiper-container swiper swiper-container-initialized swiper-container-horizontal"
    >
      <div
        class="swiper-wrapper"
        style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide"
        >
          <div
            class="single-project"
            style="background: rgb(248, 247, 246);"
          >
            <div
              class="single-project__container"
            >
              <div
                class="single-project__content-wrap"
              >
                <img
                  alt="Namba Food"
                  class="single-project__logo"
                  src="[object Object]"
                />
                 
                <h4
                  class="single-project__sub-title sub-title"
                >
                  Contactless delivery service
                </h4>
                 
                <p
                  class="single-project__paragraph paragraph"
                >
                  Mad Devs created the Namba Food delivery service from scratch. The
        apps for couriers, end users placing orders and business owners
        work as a seamless system, ensuring a smooth delivery process for
        food and other goods.
                </p>
                 
                <div
                  class="contribution-vidget"
                >
                  <img
                    alt="Mad Devs"
                    src="[object Object]"
                  />
                   
                  <span
                    class="contribution-vidget__content"
                  >
                    contribution: backend, infrastructure
                  </span>
                </div>
              </div>
               
              <div
                class="single-project__background"
              />
            </div>
          </div>
        </div>
         
        <div
          class="swiper-slide"
        >
          <div
            class="single-project"
            style="background: rgb(86, 68, 142);"
          >
            <div
              class="single-project__container"
            >
              <div
                class="single-project__content-wrap"
              >
                <img
                  alt="Namba Food"
                  class="single-project__logo"
                  src="[object Object]"
                />
                 
                <h4
                  class="single-project__sub-title sub-title"
                >
                  EdTech collaboration platform
                </h4>
                 
                <p
                  class="single-project__paragraph paragraph"
                >
                  Mad Devs improved the collaboration experience for teachers and students by the feedback-driven development of the Teacherly educational platform.
                </p>
                 
                <div
                  class="contribution-vidget"
                >
                  <img
                    alt="Mad Devs"
                    src="[object Object]"
                  />
                   
                  <span
                    class="contribution-vidget__content"
                  >
                    contribution: backend, infrastructure
                  </span>
                </div>
              </div>
               
              <div
                class="single-project__background"
              />
            </div>
          </div>
        </div>
         
        <div
          class="swiper-slide"
        >
          <div
            class="single-project"
            style="background: rgb(14, 27, 39);"
          >
            <div
              class="single-project__container"
            >
              <div
                class="single-project__content-wrap"
              >
                <img
                  alt="Namba Food"
                  class="single-project__logo"
                  src="[object Object]"
                />
                 
                <h4
                  class="single-project__sub-title sub-title"
                >
                  Cloud cybersecurity service
                </h4>
                 
                <p
                  class="single-project__paragraph paragraph"
                >
                  Mad Devs was involved with Guardrails' security check service as a development contractor with exceptional knowledge of GitHub and GitLab processes.
                </p>
                 
                <div
                  class="contribution-vidget"
                >
                  <img
                    alt="Mad Devs"
                    src="[object Object]"
                  />
                   
                  <span
                    class="contribution-vidget__content"
                  >
                    contribution: backend, infrastructure
                  </span>
                </div>
              </div>
               
              <div
                class="single-project__background"
              />
            </div>
          </div>
        </div>
         
        <div
          class="swiper-slide"
        >
          <div
            class="single-project"
            style="background: rgb(255, 106, 1);"
          >
            <div
              class="single-project__container"
            >
              <div
                class="single-project__content-wrap"
              >
                <img
                  alt="Namba Food"
                  class="single-project__logo"
                  src="[object Object]"
                />
                 
                <h4
                  class="single-project__sub-title sub-title"
                >
                  Mass transportation company
                </h4>
                 
                <p
                  class="single-project__paragraph paragraph"
                >
                  Mad Devs helped to automate bus transportation in Vietnam by creating feature-rich GoDee applications for both commuters and bus drivers.
                </p>
                 
                <div
                  class="contribution-vidget"
                >
                  <img
                    alt="Mad Devs"
                    src="[object Object]"
                  />
                   
                  <span
                    class="contribution-vidget__content"
                  >
                    contribution: backend, infrastructure
                  </span>
                </div>
              </div>
               
              <div
                class="single-project__background"
              />
            </div>
          </div>
        </div>
      </div>
      <span
        aria-atomic="true"
        aria-live="assertive"
        class="swiper-notification"
      />
    </div>
  </div>
</section>
`;
